extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/MemoryCache.java

```diff
@@ -0,0 +1,130 @@
+package io.quarkus.oidc.runtime;
+
+import java.time.Duration;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+public class MemoryCache<T> {
+    private volatile Long timerId = null;
+
+    private Map<String, CacheEntry<T>> cacheMap = new ConcurrentHashMap<>();;
+    private AtomicInteger size = new AtomicInteger();
+    private final Duration cacheTimeToLive;
+    private final int cacheSize;
+
+    public MemoryCache(Vertx vertx, Optional<Duration> cleanUpTimerInterval,
+            Duration cacheTimeToLive, int cacheSize) {
+        this.cacheTimeToLive = cacheTimeToLive;
+        this.cacheSize = cacheSize;
+        init(vertx, cleanUpTimerInterval);
+    }
+
+    private void init(Vertx vertx, Optional<Duration> cleanUpTimerInterval) {
+        cacheMap = new ConcurrentHashMap<>();
```

Comment:
`cacheMap` is initialized implicitly and it is never set to `null`, creating concurrent map  again is unnecessary and the field can be final